### ISSUE TITLE
Apache Bench (`ab`) times out if you don't manually add a Connection HTTP header

### ISSUE THREAD

**Issue by therumbler** [Kludex/starlette#128]
without manually setting the Connection HTTP header in a response, Apache Bench (Version 2.3 <$Revision: 1528965 $>) reports 5000ms + connection times. 

If you manually set the Connection HTTP header to either "Keep-Alive" or "close" then `ab` shows results as expected (1ms connection time).

e.g. 

### _without_ a valid Connection header:

```
$ ab -v 2 -c 1 -n 1 http://localhost:5044/
This is ApacheBench, Version 2.3 <$Revision: 1528965 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking localhost (be patient)...INFO: GET header ==
---
GET / HTTP/1.0
Host: localhost:5044
User-Agent: ApacheBench/2.3
Accept: */*


---
LOG: header received:
HTTP/1.1 200 OK
server: uvicorn
date: Thu, 18 Oct 2018 21:33:50 GMT
transfer-encoding: chunked

6
string
0


..done


Server Software:
Server Hostname:        localhost
Server Port:            5044

Document Path:          /
Document Length:        16 bytes

Concurrency Level:      1
Time taken for tests:   5.001 seconds
Complete requests:      1
Failed requests:        0
Total transferred:      139 bytes
HTML transferred:       16 bytes
Requests per second:    0.20 [#/sec] (mean)
Time per request:       5001.050 [ms] (mean)
Time per request:       5001.050 [ms] (mean, across all concurrent requests)
Transfer rate:          0.03 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    0   0.0      0       0
Processing:  5001 5001   0.0   5001    5001
Waiting:        1    1   0.0      1       1
Total:       5001 5001   0.0   5001    5001
$
```

### _with_ a properly set header ("close" or "Keep-Alive")

```
$ ab -v 2 -c 1 -n 1 http://localhost:5044/
This is ApacheBench, Version 2.3 <$Revision: 1528965 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking localhost (be patient)...INFO: GET header ==
---
GET / HTTP/1.0
Host: localhost:5044
User-Agent: ApacheBench/2.3
Accept: */*


---
LOG: header received:
HTTP/1.1 200 OK
server: uvicorn
date: Thu, 18 Oct 2018 21:39:02 GMT
connection: close
transfer-encoding: chunked

6
string
0


..done


Server Software:
Server Hostname:        localhost
Server Port:            5044

Document Path:          /
Document Length:        16 bytes

Concurrency Level:      1
Time taken for tests:   0.001 seconds
Complete requests:      1
Failed requests:        0
Total transferred:      136 bytes
HTML transferred:       16 bytes
Requests per second:    677.05 [#/sec] (mean)
Time per request:       1.477 [ms] (mean)
Time per request:       1.477 [ms] (mean, across all concurrent requests)
Transfer rate:          89.92 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    0   0.0      0       0
Processing:     1    1   0.0      1       1
Waiting:        1    1   0.0      1       1
Total:          1    1   0.0      1       1
```

code:

```
from starlette.applications import Starlette
from starlette.responses import JSONResponse, Response

import uvicorn

app = Starlette()

@app.route('/')
async def index(request):
    return Response('string', headers={'Connection': 'close'})

if __name__ == '__main__':
    uvicorn.run(app, host='0.0.0.0', port=5044)
```

**Comment by lovelydinosaur** [comment]
So, this is due to a difference in HTTP/1.0 and HTTP/1.1 behaviour... https://stackoverflow.com/questions/17437950/when-does-an-http-1-0-server-close-the-connection

Uvicorn is currently using http/1.1 behaviour (connections default to keep-alive) where it should be checking for http/1.0 or http/1.1 and adjusting its default connection keep alive/close accordingly.

**Comment by lovelydinosaur** [comment]
Believe this to now be resolved in Uvicorn 0.3.14.
